bubble_sort.rb

```diff
@@ -0,0 +1,33 @@
+def bubble_sort(arr)
+	def swap(val1, val2, arr, index)
+		arr[index+1] = val1
+		arr[index] = val2
+	end
+	n = arr.length-1
+	k = n
+	n.times do
+		arr.each_with_index do |value, index|
+			break if index==k
+			if arr[index] > arr[index+1]
+				swap(arr[index], arr[index+1], arr, index)
```

Comment:
Ruby has a neat way to do swaps: 
`arr[index], arr[index + 1] = arr[index + 1], arr[index]`
That way you don't need to define your auxiliar method :)